planner/core/common_plans.go

```diff
@@ -550,12 +550,12 @@ type LoadData struct {
 	FileLocRef  ast.FileLocRefTp
 	OnDuplicate ast.OnDuplicateKeyHandlingType
 	Path        string
-	Format      string
+	Format      *string
 	Table       *ast.TableName
 	Columns     []*ast.ColumnName
 	FieldsInfo  *ast.FieldsClause
 	LinesInfo   *ast.LinesClause
-	NullInfo    *ast.NullDefinedBy
+	LineFieldsInfo
```

Comment:
why do we need both `LineFieldsInfo` and `FieldsInfo`/`LinesInfo` 😵

Reply:
DefinedNullBy is inited in loadDataController

Reply:
Seems `LineFieldsInfo` is the common part of load-data, select-into and index advise. And for LOAD DATA we need extra FieldsInfo to handle NULL config, and extra FieldsInfo to check with Format.

We don't need to add LineFieldsInfo to the "LoadData plan" to avoid repeating? LineFieldsInfo can be build later from plan.FieldsInfo/LinesInfo in loadDataController

Reply:
https://github.com/pingcap/tidb/pull/42191/commits/5aeaebebe9642f16e01119f7c1b97ef7be422c20